connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala

```diff
@@ -386,143 +365,152 @@ object KustoWriter {
-        // Need to copy the ingestionProperties so that only this blob ingestion will be effected
-        props = SparkIngestionProperties.cloneIngestionProperties(ingestionProperties)
-      }
-
-      if (parameters.writeOptions.ensureNoDupBlobs) {
-        val pref = KDSU.getDedupTagsPrefix(parameters.writeOptions.requestId, batchIdForTracing)
-        val tag = pref + blobUUID
-        val ingestIfNotExist = new util.ArrayList[String]
-        ingestIfNotExist.addAll(props.getIngestIfNotExists)
-        val ingestBy: util.List[String] = new util.ArrayList[String]
-        ingestBy.addAll(props.getIngestByTags)
-
-        ingestBy.add(tag)
-        ingestIfNotExist.add(tag)
-        props.setIngestByTags(ingestBy)
-        props.setIngestIfNotExists(ingestIfNotExist)
-      }
-
-      if (!props.getFlushImmediately && flushImmediately) {
-        props.setFlushImmediately(true)
-      }
-      // write the data here
-      val partitionsResult = KDSU.retryApplyFunction(
-        () => {
-          Try(
-            ingestClient.ingestFromBlob(
-              new BlobSourceInfo(blobUri + sas, size, UUID.randomUUID()),
-              props)) match {
-            case Success(x) =>
-              <!-- The statuses of the ingestion operations are now set in the ingestion result -->
-              val blobUrlWithSas =
-                s"${blobResource.blob.getStorageUri.getPrimaryUri.toString}${blobResource.sas}"
-              val containerWithSas = new ContainerWithSas(blobUrlWithSas, null)
-              kustoClient.reportIngestionResult(containerWithSas, success = true)
-              x
-            case Failure(e: Throwable) =>
-              KDSU.reportExceptionAndThrow(
-                className,
-                e,
-                "Queueing blob for ingestion in partition " +
-                  s"$partitionIdString for requestId: '${parameters.writeOptions.requestId}")
-              val blobUrlWithSas =
-                s"${blobResource.blob.getStorageUri.getPrimaryUri.toString}${blobResource.sas}"
-              val containerWithSas = new ContainerWithSas(blobUrlWithSas, null)
-              kustoClient.reportIngestionResult(containerWithSas, success = false)
-              null
-          }
-        },
-        this.retryConfig,
-        "Ingest into Kusto")
-      if (parameters.writeOptions.isTransactionalMode) {
-        partitionsResults.add(PartitionResult(partitionsResult, partitionId))
-      }
-      KDSU.logInfo(
-        className,
-        s"Queued blob for ingestion in partition $partitionIdString " +
-          s"for requestId: '${parameters.writeOptions.requestId}")
-    }
     val kustoClient = KustoClientCache.getClient(
       parameters.coordinates.clusterUrl,
       parameters.authentication,
       parameters.coordinates.ingestionUrl,
       parameters.coordinates.clusterAlias)
+    val blobIdMap = new ConcurrentHashMap[String, Int]()
```

Comment:
did you mean for this to be some static cache ? if so it will need to live outside the function as a member.
currently this will always hold only the current run with the same request id in all Map entries, 
Regardless - a simple integer with ncrements would do the job i think